packages/core/test/history.test.ts

```diff
@@ -1368,3 +1369,142 @@ describe('multistage history states', () => {
     });
   });
 });
+
+// --- Tests for reviveHistoryValue ---
+describe('revive history states', () => {
+  const {
+    restoreSnapshot,
+    getInitialSnapshot
+  } = require('../src/StateMachine');
```

Comment:
1. we shouldnt use `require` in tests
2. we should focus on testing behaviors thought public facing APIs alone

Reply:
1. was autoimport mistake
2. I have refactored tests to public only API